crates/wast/src/component/types.rs

```diff
@@ -271,6 +268,31 @@ impl<'a> Parse<'a> for PrimitiveValType {
     }
 }
 
+impl Peek for PrimitiveValType {
```

Comment:
Is this and `Peek for ComponentDefinedType` necessary to define? I find it difficult to keep `Peek` and `Parse` in sync so try to leave out `Peek` unless it's necessary

Reply:
The parsing of `ResultType` has an optional `ComponentValType`, so this and the implementation for `ComponentDefinedType` were needed to prevent the implementation of `ResultType` from being very complex.

I realize that `Peek` has a bit of a maintenance overhead since 1) it deals with strings rather than `kw` and 2) it's easy to drift from the parse implementation.

I'm definitely open to alternative approaches, though.

Reply:
Oh right yeah that makes sense. An nah as long as it's necessary I think it's fine to have!